src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -16,13 +16,23 @@ type KueryNode = any;
 
 const astFunctionType = ['is', 'range', 'nested'];
 
+const addAttributesInMappingIndex = (indexMapping: IndexMapping) => {
+  if (!Object.keys(indexMapping.properties).includes('_id')) {
+    indexMapping.properties = {
+      _id: { type: 'keyword' },
```

Comment:
We need to add the `_id` since it is not defined in our `defaultMapping` and I think that's a good place since we just want to allow that for filtering.

https://github.com/elastic/kibana/blob/8.1/src/core/server/saved_objects/migrations/core/build_active_mappings.ts#L117

Reply:
You're mutating the mapping passed down to `validateConvertFilterToKueryNode` by side effect:

https://github.com/elastic/kibana/blob/6627bd8b3ab49415e397495d8e076f9861fcc894/src/core/server/saved_objects/service/lib/repository.ts#L965

Which can have impacts on other part of the code relying on these mappings.

I would just return a new object instead of doing that:

```ts
const addPropertiesInMappingIndex = (indexMapping: IndexMapping) => {
  return {
     ...indexMapping,
     properties: {
         _id: { type: 'keyword' },
         ...indexMapping.properties,
     },
  };
};
```

Reply:
Saved Objects have `id` properties unlike the Elasticsearch document which has `_id`. So to keep the API consistent we should ideally rewrite `id` into an `_id` filter. (this is why we need KPT 😉 🐣 )

Reply:
So your example would become:
```
curl --location --request GET 'http://localhost:5601/kbn/api/detection_engine/rules/_find?page=1&sort_field=enabled&sort_order=desc&filter=alert.id: "alert:8e5a93a0-9320-11ec-9265-8b772383a08d"&per_page=100' \
--header 'kbn-xsrf: reporting' \
--header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
--data-raw ''
```